libnetwork/firewall_linux_test.go

```diff
@@ -96,10 +105,13 @@ func getRules(t *testing.T, chain string) []string {
 }
 
 func resetIptables(t *testing.T) {
-	iptable := iptables.GetIptable(iptables.IPv4)
-
 	t.Helper()
-	_, err := iptable.Raw("-F", fwdChainName)
-	assert.NilError(t, err)
-	_ = iptable.RemoveExistingChain(usrChainName, "")
+
+	for _, ipVer := range []iptables.IPVersion{iptables.IPv4, iptables.IPv6} {
+		iptable := iptables.GetIptable(ipVer)
+
+		_, err := iptable.Raw("-F", fwdChainName)
+		assert.NilError(t, err)
```

Comment:
Should the test be aborted here, without cleaning up the user chain?
```suggestion
		assert.Check(t, err)
```

Reply:
Right, cleanup should happen for IPv6 rules even when it can't be done for IPv4. However, this cleanup is quite messy: 1. flush is also done by `RemoveExistingChain`, and 2. this method calls `*ChainInfo.Remove()`, which never returns an error. Would be better to simply remove the `Raw()` call here and let `RemoveExistingChain` properly report any error. It's out of scope for this PR, but would be great to improve it whenever we do some refactoring.